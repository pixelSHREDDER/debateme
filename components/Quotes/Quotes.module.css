.quotes div {
  color: var(--mantine-color-text);
  font-size: 0.8rem;
}
