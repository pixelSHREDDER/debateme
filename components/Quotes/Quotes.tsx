import { theme } from '@/theme'
import { Alert } from '@mantine/core'
import { useId } from 'react'
import quotesStyles from './Quotes.module.css'
import quotesText from './QuotesText'

export default function Quotes() {
  const id = useId()
  const getQuote = () => quotesText[Math.floor(Math.random() * quotesText.length)]
  return (
    <Alert
      className={quotesStyles.quotes}
      component="aside"
      variant="outline"
      fs="italic"
      id={`alert_quotes_${id}`}
      p={6}
      color={theme.colors?.purple?.[4] || 'purple'}
      radius="md"
      w="auto"
      maw="75%">
      {getQuote()}
    </Alert>
  )
}
