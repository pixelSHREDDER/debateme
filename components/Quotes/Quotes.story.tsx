import type { Meta, StoryObj } from '@storybook/react'
import Quotes from './Quotes'

const meta: Meta<typeof Quotes> = {
  title: 'Quotes',
  component: Quotes,
}

export default meta

type Story = StoryObj<typeof Quotes>

export const Quote: Story = {}
