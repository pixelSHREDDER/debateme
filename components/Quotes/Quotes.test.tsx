import { describe, expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import ProviderWrapper from '@/tests/helpers/ProviderWrapper'
import Quotes from './Quotes'

describe('Quotes component', () => {
  test('renders correctly', () => {
    render(
      <Quotes />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByRole('alert')).toMatchSnapshot()
  })
})
