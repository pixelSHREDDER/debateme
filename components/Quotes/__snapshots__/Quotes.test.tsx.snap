// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Quotes component > renders correctly 1`] = `
<aside
  aria-describedby="alert_quotes_:r0:-body"
  class="_quotes_e6219c m_66836ed3 mantine-Alert-root"
  data-variant="outline"
  id="alert_quotes_:r0:"
  role="alert"
  style="--alert-radius: var(--mantine-radius-md); --alert-bg: transparent; --alert-color: #9562d2; --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid #9562d2; font-style: italic; width: auto; max-width: 75%;"
>
  <div
    class="m_a5d60502 mantine-Alert-wrapper"
  >
    <div
      class="m_667c2793 mantine-Alert-body"
    >
      <div
        class="m_7fa78076 mantine-Alert-message"
        data-variant="outline"
        id="alert_quotes_:r0:-body"
      >
        “Don’t be a dick.” ― Unknown
      </div>
    </div>
  </div>
</aside>
`;
